projects/storefrontapp-e2e-cypress/cypress/helpers/asm.ts

```diff
@@ -149,6 +149,35 @@ export function asmTests() {
         cy.get('cx-customer-selection').should('not.exist');
       });
 
+      it('should log in as an ASM agent, emulate customer and end the session with the end session button.', () => {
+        // should login as ASM agent
+        const authenticationRequestAlias = asm.listenForAuthenticationRequest();
+        cy.get('cx-csagent-login-form').should('exist');
+        cy.get('cx-customer-selection').should('not.exist');
+        cy.get('input[formcontrolname="userId"]').type('asagent');
+        cy.get('input[formcontrolname="password"]').type('123456');
+        cy.get('button[type="submit"]').click();
+
+        cy.wait(authenticationRequestAlias)
+          .its('status')
+          .should('eq', 200);
+
+        // should emulate a customer
+        cy.get('cx-csagent-login-form').should('not.exist');
+        cy.get('cx-customer-selection').should('exist');
+
+        cy.get('cx-customer-selection form input').type(customer.email);
+        cy.get('cx-customer-selection form button').click();
+
+        cy.get('cx-customer-selection').should('not.exist');
+        cy.get('div.cx-customer-emulation').should('exist');
+
+        // should end a user's session when clicking on the end session button
+        cy.get('div.cx-customer-emulation button').click();
+        cy.get('div.cx-customer-emulation').should('not.exist');
+        cy.get('cx-customer-selection').should('exist');
+      });
+
```

Comment:
The test for the end session button should rather be after we end the session ( test 'agent should stop customer emulation.' ) but before we sign out the agent.
This way the test will be simpler.  We only need to start a customer emulation and end it right away.